src/inc/Vibrato.h

```diff
@@ -1,4 +1,95 @@
 #if !defined(__Vibrato_hdr__)
 #define __Vibrato_hdr__
 
+#include "ErrorDef.h"
+#include "Lfo.h"
+#include "RingBuffer.h"
+
+/*! \brief A vibrato using wavetable LFO
+* 
+* This vibrato is single channel only, to keep the implementation simple, since
+* having a multichannel vibrato but keep every parameter in each channel the same 
+* don't have much value. If the user wants multichannel vibrato, they can just 
+* have one instance of this class for each channels.
+*  
+* This interface is designed (100% copied) from the previous CCombFilterIf and
+* AudioFileIf classes to keep the repository in style.
+* The create and destroy function and the protected constructor and destructor
+* are probably designed to provide extra memory management to the user. However, 
+* I haven't looked into when the program throw an error and run out of the scope
+* of an instance, would this give memory issue or not. Modern C++ has things like
+* smart pointers to manage memories, don't know if it would be better to use that.
```

Comment:
Destructors should still get called regardless of how you exit the scope.
Smart pointers are nice; `std::unique_ptr`, for instance, makes it harder to accidentally leak memory by tying allocation/deallocation to scope (as occurs automatically for variables on the stack).
IMO, it's even better if you can avoid unnecessary pointers (and thus indirection + potential leaks) altogether. In this case, there is no real need for the LFO and RingBuffer to be allocated separately on the heap; the only reason is because the Vibrato's construction and initialization have been split up (into the constructor and a separate `init()` method). (This also means CVibrato can be in an awkward "created but unusable" state which doesn't afflict your CWavetableLFO.)
But I appreciate that you're following the conventions of other classes in the repo.